src/neo/Ledger/MemoryPool.cs

```diff
@@ -95,10 +95,11 @@ public int Count
 
         public int UnVerifiedCount => _unverifiedTransactions.Count;
 
-        public MemoryPool(NeoSystem system)
+        public MemoryPool(NeoSystem system) : this(system, system.Settings.MemoryPoolMaxTransactions) { }
+        public MemoryPool(NeoSystem system, int capacity)
```

Comment:
Why?

Reply:
Otherwise I need to make public the `ProtocolSettings` constructor to test MemPool with different values.

Reply:
Maybe we can keep `capacity` and give it a default value `system.Settings.MemoryPoolMaxTransactions`

Reply:
@shargon Now `ProtocolSettings` has become a record. You can use `with` expression to create new settings.

```c#
ProtocolSettings settings = ProtocolSettings.Default with { MemoryPoolMaxTransactions = 10 };
```